go/vt/topo/keyspace.go

```diff
@@ -191,9 +200,50 @@ func (ts *Server) FindAllShardsInKeyspace(ctx context.Context, keyspace string,
 		opt.Concurrency = 1
 	}
 
+	// First try to get all shards using List if we can.
+	shardsPath := path.Join(KeyspacesPath, keyspace, ShardsPath)
+	listResults, err := ts.globalCell.List(ctx, shardsPath)
+	if err != nil || len(listResults) == 0 {
```

Comment:
I am confused why the `len(listResults) == 0` is required here. If we do not have an error but we have an empty result set, don't we want to go into the `else` block?

Reply:
or generally, don't we always want to go into the `else` block if we don't have any error?